### PR TITLE
roleIDs into context helper + roles cache + roles cache update middleware

### PR DESCRIPTION
**Pull request by kulmann**

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- roles/cache.go  5
- middleware/roles.go  7
- roles/option.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-pkg/pullRequest?prid=6076817&bid=19894840)

File: go.mod [hunk 1]
```diff
@@ -23,3 +24,5 @@ require (
 )
 
 replace google.golang.org/grpc => google.golang.org/grpc v1.26.0
+
+replace github.com/owncloud/ocis-settings => ../ocis-settings
```
Comment (butonic) [inline go.mod]: waiting for https://github.com/owncloud/ocis-settings/pull/59

File: middleware/roles.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	"encoding/json"
+	"net/http"
+
+	"github.com/micro/go-micro/v2/metadata"
+	"github.com/owncloud/ocis-pkg/v2/log"
+	"github.com/owncloud/ocis-pkg/v2/roles"
+	settings "github.com/owncloud/ocis-settings/pkg/proto/v0"
+)
+
+func Roles(log log.Logger, rs settings.RoleService, cache *roles.Cache) func(next http.Handler) http.Handler {
```
Comment (butonic) [inline middleware/roles.go:16]: add comment what does the Roles middleware does

File: roles/cache.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	settings "github.com/owncloud/ocis-settings/pkg/proto/v0"
+)
+
+// entry extends a bundle and adds a TTL
+type entry struct {
+	*settings.Bundle
+	inserted time.Time
+}
+
+// Cache is a barebones cache implementation.
```
Comment (butonic) [inline roles/cache.go]: implements a roleid -> role cache

**Review by butonic — APPROVED** [review]
when you update the go.mod also add the missing comments

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- roles/cache.go  5
- middleware/roles.go  7
- roles/option.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-pkg/pullRequest?prid=6076817&bid=19894840)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- roles/cache.go  5
- middleware/roles.go  7
- roles/option.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-pkg/pullRequest?prid=6076817&bid=19894840)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- roles/cache.go  5
- middleware/roles.go  7
- roles/option.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-pkg/pullRequest?prid=6076817&bid=19894840)

**Pull request merged**